src/Microsoft.Fx.Portability.MetadataReader/ReflectionMetadataDependencyInfo.cs

```diff
@@ -196,5 +197,33 @@ public InvalidPEAssemblyException(Exception inner)
                 : base("Not a valid assembly", inner)
             { }
         }
+
```

Comment:
Yup - I prototyped that as well, but then you have to deal with Interlocked.Exchange or something to ensure the updated set is swapped with the original since immutable collections don't update. I went with this as we're already using ConcurrentDictionaries and it was a fairly simple swap.